compiler/internal/codegen/codegen_main.go

```diff
@@ -113,6 +123,49 @@ func (b *Builder) computeStaticPubsubConfig() Code {
 	return Map(String()).Op("*").Qual("encore.dev/appruntime/config", "StaticPubsubTopic").Values(pubsubTopicDict)
 }
 
+func (b *Builder) computeCORSHeaders() (Code, error) {
+	// Find all used headers
+	usedHeaderSet := map[string]struct{}{}
+	usedHeaders := []string{}
+
+	// Walk all the structures looking for headers
+	decls := b.res.Meta.Decls
+	for _, decl := range decls {
```

Comment:
Better to start from APIs, and only check request structs for non-private APIs? Otherwise this will end up adding a bunch of extraneous headers (such as for private endpoints, response headers, etc)

Reply:
What about middleware?

That's why I started at decls

Reply:
Middleware cannot add headers I don't believe? Auth handlers can so they need to be checked also

Reply:
I meant Auth handlers :P